latex_engine_cmds.bzl

```diff
@@ -201,71 +35,80 @@ def add_flags(ctx):
         assumption that it is the luahbtex binary (This should be
         enforced more strongly).
 
     Returns:
-      A list with the three commands
+      A list of the commands
     """
 
-    # TODO: Each command could be its own invokation to ctx.actions.run
-    #       this way the artifacts could be cached, it might not help that
-    #       much though, as any change to the inputs is likely to re-trigger
-    #       a re-build.
-    #       However, running each command by ctx.actions.run will off-load
-    #       the tool_wrapper, so that it can be removed.
-    int_files = ctx.files.font_maps + ctx.files.ini_files + ctx.files.web2c
-    files = ctx.files.srcs + int_files
-
-    ini_args = get_common_args(ctx, dep_tools, files)
+    fmt = ctx.actions.declare_file(
+        ctx.label.name + "/" + ctx.attr._progname + ".fmt",
+    )
+    ini_args = ctx.actions.args()
     ini_files_path = ctx.files.ini_files[0].dirname
-    engine_config_args = "{}/{}.ini".format(ini_files_path, ctx.attr._progname)
-    ini_args.add("--input", engine_config_args)
-    ini_args.add("--tool", latex_tool.to_list()[0])
-    ini_args.add("--tool-output", "{}.fmt".format(ctx.attr._progname))
-    ini_args.add("--flag=-ini")
-
-    ini_args.use_param_file("--file=%s", use_always = True)
-    ini_args.set_param_file_format("multiline")
+    ini_args.add("-ini")
+    ini_args.add(
+        "--output-directory",
+        ctx.outputs.out.dirname + "/" + ctx.label.name,
+    )
+    ini_args.add("{}/{}.ini".format(ini_files_path, ctx.attr._progname))
 
-    env_args = get_common_args(ctx, dep_tools, files)
-    env_args.add("--tool", dep_tools[4].to_list()[0].path)
-    env_args.add("--input", ".")
-    env_args.use_param_file("--file=%s", use_always = True)
-    env_args.set_param_file_format("multiline")
+    # TODO: Add usage of ls-R database to possibly reduce search times
+    # ls_r = ctx.actions.declare_file("ls-R")
+    # env_args = ctx.actions.args()
+    # env_args.add(dep_tools[4].to_list()[0].path)
+    # env_args.add(".")
 
-    args = get_common_args(ctx, dep_tools + [latex_tool], files)
-    args.add("--tool=" + ctx.files._latexrun[0].path)
+    args = ctx.actions.args()
     args.add(
-        "--flag=--latex-cmd={}".format(
-            latex_tool.to_list()[0].path,
+        "--latex-cmd={}".format(
+            latex_tool.files.to_list()[0].path,
         ),
     )
-    args.add("--flag=-Wall")
-    args.add("--flag=--debug")
-    args.add("--input=" + ctx.file.main.path)
-    args.add("--tool-output=" + ctx.label.name + ".{}".format(ctx.attr.format))
-    args.add("--output=" + ctx.outputs.out.path)
-    args.add("--flag=-O=.")
-    args.add_all(add_flags(ctx))
-
-    args.use_param_file("--file=%s", use_always = True)
-    args.set_param_file_format("multiline")
-
-    return [ini_args, env_args, args]
+    args.add("-Wall")
+
+    # TODO: Add debug flags
+    # args.add("--debug")
```

Comment:
The same `config_setting` could be used here

Reply:
Let me think about it the debug output doesn't really affect the artifact output.

Reply:
removing the debug TODOs for now, I can write an issue about it later and we can discuss pros- and cons there.

Reply:
Yeah, maybe you are right. 😊 But I'll add it for another pull request